sim/battle-actions.ts

```diff
@@ -1435,7 +1435,7 @@ export class BattleActions {
 				this.battle.heal(pokemon.maxhp, pokemon, pokemon, zPower);
 				break;
 			case 'healreplacement':
-				move.self = {slotCondition: 'healreplacement'};
+				pokemon.side.addSlotCondition(pokemon, 'healreplacement', pokemon);
```

Comment:
I'm not sure whether this will set the source effect correctly, so you might need to append `, move` here.

Reply:
Good catch!